### ISSUE TITLE
[Enhancement]: rel=nofollow on Blocks

### ISSUE THREAD

**Issue by SirDarcanos** [woothemes/woocommerce#43097]
### Describe the solution you'd like

Similar to https://github.com/woocommerce/woocommerce/issues/41190, the Gutenberg blocks add a `rel="nofollow"` attribute to all Add to Cart links, even if they are variable products where they just link to the single product page.

These buttons should not have a nofollow directive as they are internal links, and they literally point to another page of the site.

### Describe alternatives you've considered

There's no alternative to this. Filters do not work, nor you can use a robot.txt like in the other issue since there's no parameter in the URL. It's just a plain URL to the product page.,. It'scan you

**Comment by Aljullu** [comment]
Thanks for opening this issue, @SiR-DanieL!

@kmanijak @imanish003 I wonder if this is something that we could implement in the Product Collection block? We are setting `rel="nofollow"` here:

https://github.com/woocommerce/woocommerce/blob/bffe12b9451e9fac4886a0a8bb5d47dd1f571b0d/plugins/woocommerce/src/Blocks/BlockTypes/ProductButton.php#L153

Based on what's exposed in https://github.com/woocommerce/woocommerce/issues/41190, I don't think we should remove `ref="nofollow"` from the _Add to Cart_ of all products as it can have an impact on SEO. But I think it's safe to remove it if:

* `! $product->is_purchasable() || ! $product->is_in_stock()`: if the product can't be purchased, the _Add to Cart_ button is replaced with a link to the product page, in which case it should be ok to remove the `rel="nofollow"`. That would take into consideration variable, grouped, out of stock and external products.

For simple products which are in stock, I think we should wait until https://github.com/woocommerce/woocommerce/issues/41190 is fixed first. Thoughts?

**Comment by imanish003** [comment]
Hi @Aljullu,
I agree with you. However, I'm trying to understand how the Product Collection block might implement this, as it appears to be a functionality intrinsic to the Product Button. Perhaps I'm overlooking an aspect? Could you please elaborate a bit further? Shouldn't it be fixed in Product Button?

**Comment by Aljullu** [comment]
Yes, sorry. I mentioned the Product Collection block as it was the wrapper block that I first thought about. :sweat_smile: 

But you are right, these changes would need to be implemented in the Product Button block, so they will also impact the Products (Beta) block and the Single Product block if I'm not wrong.

**Comment by imanish003** [comment]
@Aljullu, no problem at all! It seems like a minor task. Since it's not directly tied to any ongoing projects, perhaps Porter could handle this one. What do you think?

**Comment by Aljullu** [comment]
Yeah, I don't have a strong opinion. :slightly_smiling_face: I guess we can leave it in the backlog in case somebody (or the porter) wants to take it. I also added the `good first issue` label. With the [instructions from my previous comment](https://github.com/woocommerce/woocommerce/issues/43097#issuecomment-1877179376), I think it should be easy to implement.

**Comment by kkoteen** [comment]
Reviewed and confirmed this is still valid. Marking as ready for engineering.

@Aljullu moving this back to you. I'm also going to close the open PR #43432 since there was no response from the contributor.
